.github/workflows/installer-build-check.yml

```diff
@@ -1,76 +1,129 @@
 name: Build Test Installers
 
-# Job will run nightly at 03:05 EDT / 02:05 CDT
-# Time below is UTC
 on:
-  schedule:
-    - cron: "5 7 * * *"
   workflow_dispatch:
   push:
     branches:
       - gha-scheduled-*
 
-
 jobs:
   build-installers:
     name: Build Installers
-    runs-on: macos-latest #needs macos for apple notarization
+    runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v4
 
       - name: Set up JDK for GPG
         uses: actions/setup-java@v4
         with:
-          java-version: '17'
-          distribution: 'temurin'
+          java-version: "17"
+          distribution: "temurin"
           gpg-private-key: ${{ secrets.GPG_SECRET }}
           gpg-passphrase: GPG_PASSPHRASE
           overwrite-settings: false
         env:
           GPG_PASSWORD: ${{ secrets.GPG_PASSPHRASE }}
 
+      # TODO: Add install4j and DigiCert KeyLocker Client Tools caching if possible
+      # - name: Cache Install4j
+      #   id: cache-install4j
+      #   uses: actions/cache@v4
+      #   with:
+      #     path: /usr/local/bin/install4jc
+      #     key: install4j-${{ runner.os }}-10_0_9
+
+      # - name: Install Install4j (if not cached)
+      #   if: steps.cache-install4j.outputs.cache-hit != 'true'
+      #   run: |
+      #     wget -nv -O install4j.deb "https://download.ej-technologies.com/install4j/install4j_linux-x64_10_0_9.deb"
+      #     sudo apt install -y ./install4j.deb
+      #     rm install4j.deb
+
+      - name: Decode Client Authentication Certificate
+        run: |
+          echo "${{ secrets.SM_CLIENT_CERT_FILE_B64 }}" | base64 --decode > /tmp/client-auth.p12
+
+      - name: Set Environment Variables for Signing
+        run: |
+          echo "SM_HOST=${{ secrets.SM_HOST }}" >> "$GITHUB_ENV"
+          echo "SM_API_KEY=${{ secrets.SM_API_KEY }}" >> "$GITHUB_ENV"
+          echo "SM_CLIENT_CERT_FILE=/tmp/client-auth.p12" >> "$GITHUB_ENV"
+          echo "SM_CLIENT_CERT_PASSWORD=${{ secrets.SM_CLIENT_CERT_PASSWORD }}" >> "$GITHUB_ENV"
+          echo "SM_CODE_SIGNING_CERT_SHA1_HASH=${{ secrets.SM_CODE_SIGNING_CERT_SHA1_HASH }}" >> "$GITHUB_ENV"
+          echo "SM_KEY_PAIR_ALIAS=${{ secrets.SM_KEY_PAIR_ALIAS }}" >> "$GITHUB_ENV"
+
+      - name: Install jsign and osslsigncode
+        run: |
+          curl -fSslL https://github.com/ebourg/jsign/releases/download/3.1/jsign_3.1_all.deb -o jsign_3.1_all.deb
+          sudo dpkg --install jsign_3.1_all.deb
+          sudo apt-get install osslsigncode
+
+      - name: Create pkcs11properties.cfg
+        run: |
+          cat << EOF > pkcs11properties.cfg
+          name=signingmanager
+          library="$(pwd)/.github/util/smpkcs11.so"
+          slotListIndex=0
+          EOF
+          mv pkcs11properties.cfg $(pwd)/.github/util/
+
+      - name: Install DigiCert KeyLocker Client Tools
+        uses: digicert/ssm-code-signing@v1.0.0
```

Comment:
@jandroav I am confused about what this even does and why we had to install jsign separately